streamlit_app.py

```diff
@@ -61,6 +84,10 @@ def generate_arctic_response():
 
 # User-provided prompt
 if prompt := st.chat_input(disabled=not replicate_api):
```

Comment:
Interesting. Thank you @LukasMasuch . Actually I realized that we need to avoid the case where the entire conversation gets too long. So I moved the test.